contracts/managers/allocators/SocialAllocator.sol

```diff
@@ -0,0 +1,401 @@
+     *
+     * And Pa1 is price of asset 1 and Pa2 is price of asset 2
+     *
+     * This results in a Set Token that's value is max(Pa1, Pa2) * (a1Multiplier + a2Multiplier)
+     *
+     * @param  _targetBaseAssetAllocation       Target allocation of the base asset
+     * @param  _allocationPrecision             Precision of allocation percentage
+     * @return uint256[]                        Unit array of nextSet collateral
+     * @return address[]                        Component array of nextSet collateral
+     */
+    function calculateMultiAssetNextSetUnits(
+        uint256 _targetBaseAssetAllocation,
+        uint256 _allocationPrecision
+    )
+        internal
+        view
+        returns (uint256[] memory, address[] memory)
+    {
+        // Get quote asset allocation
+        uint256 quoteAssetAllocation = _allocationPrecision.sub(_targetBaseAssetAllocation);
+
+        // Calculate multiplier for quote and base asset. Multiplier is just the amount of highest
+        // allocation divided by lowest allocation. Asset that has lowest allocation will have
+        // multiplier set to 1.
+        uint256 baseAssetMultiplier = Math.max(_targetBaseAssetAllocation.scale().div(quoteAssetAllocation), ONE.scale());
+        uint256 quoteAssetMultiplier = Math.max(quoteAssetAllocation.scale().div(_targetBaseAssetAllocation), ONE.scale());
+
+        // Get prices
+        uint256 baseAssetPrice = baseAssetOracle.read();
+        uint256 quoteAssetPrice = quoteAssetOracle.read();
+
+        uint256[] memory units = new uint256[](2);
+        // Get baseAsset units
+        units[0] = calculateUnitAmount(
+            Math.max(quoteAssetPrice.mul(pricePrecision).div(baseAssetPrice), pricePrecision),
+            baseAssetMultiplier,
+            baseAssetDecimalDifference
+        );
+
+        // Get quote asset units
+        units[1] = calculateUnitAmount(
+            Math.max(baseAssetPrice.mul(pricePrecision).div(quoteAssetPrice), pricePrecision),
+            quoteAssetMultiplier,
+            quoteAssetDecimalDifference
+        );
+
+        return (units, multiAssetNextSetComponents);
+    }
+
+    /*
+     * Calculate units of next Set. Determine which asset to rebalance into and then calculate unit
+     * amount. Set value should be equal to SINGLE_ASSET_MULTIPLIER * max(Pa1, Pa2).
+     *
+     * @param  _targetBaseAssetAllocation       Target allocation of the base asset
+     * @param  _allocationPrecision             Precision of allocation percentage
+     * @return uint256[]                        Unit array of nextSet collateral
+     * @return address[]                        Component array of nextSet collateral
+     */
+    function calculateSingleAssetNextSetUnits(
+        uint256 _targetBaseAssetAllocation,
+        uint256 _allocationPrecision
+    )
+        internal
+        view
+        returns (uint256[] memory, address[] memory)
+    {
+        // Get prices
```

Comment:
To save gas, these getPrices functions could be in the if statement

Reply:
How does that save gas? They're used in both instances.